apps/system/js/screen_manager.js

```diff
@@ -292,11 +319,15 @@ var ScreenManager = {
         break;
 
       case 'callschanged':
-        if (!telephony.calls.length &&
-            !(telephony.conferenceGroup &&
-              telephony.conferenceGroup.calls.length)) {
-
-          this.turnScreenOn();
+        if (numOpenLines() === 0) {
+          if (this._turnOffWhenCallsEnd) {
+            this._turnOffWhenCallsEnd = false;
+            setTimeout(function() {
```

Comment:
That's an UX problem: when we hangup the last call we usually display the last call on the callscreen with the duration of the call, etc... for a couple of seconds, then we hide the callscreen. If we turn off the screen immediately this cannot be displayed, the user will not see the calls being terminated. Since I didn't have an UX for this bug I went for waiting 2 seconds before turning off the screen as that's the same behavior I see on iOS (haven't tried on Android though).